src/components/tables/RowActions/AccessGrants/DeleteButton.tsx

```diff
@@ -43,49 +44,42 @@ function DeleteButton({ selectTableStoreName }: Props) {
 
     const handlers = {
         showConfirmationDialog: () => {
-            const grants: { id: string; message: string }[] = [];
+            const grants: AccessGrantRowConfirmation[] = [];
 
             selectedRows.forEach((value, _key) => {
-                if (
+                const identifier =
                     selectTableStoreName ===
                     SelectTableStoreNames.ACCESS_GRANTS_USERS
-                ) {
-                    const identifier: string =
-                        value?.user_full_name ??
-                        value?.user_email ??
-                        value?.subject_role ??
-                        'User';
+                        ? value?.user_full_name ??
+                          value?.user_email ??
+                          value?.subject_role ??
+                          'User'
+                        : value.subject_role;
 
-                    grants.push({
-                        id: value.id,
-                        message: intl.formatMessage(
-                            { id: 'admin.users.confirmation.listItem' },
-                            { identifier, capability: value.capability }
-                        ),
-                    });
-                } else {
-                    grants.push({
-                        id: value.id,
-                        message: intl.formatMessage(
-                            { id: 'admin.prefix.confirmation.listItem' },
-                            {
-                                subjectRole: value.subject_role,
-                                capability: value.capability,
-                                objectRole: value.object_role,
-                            }
-                        ),
-                    });
-                }
+                grants.push({
+                    id: value.id,
+                    details: describeAllRemovals(value),
+                    message: (
+                        <GrantWhatIsChanging
+                            capability={value.capability}
+                            identifier={identifier}
+                            grantScope={value.object_role}
+                        />
+                    ),
+                });
```

Comment:
I am hoping these can stay in sync and we don't have to split them back out again. I kinda just forced them into this for now... but there is a solid chance we'll need to split.